src/java.base/share/classes/jdk/internal/math/FloatToDecimal.java

```diff
@@ -466,37 +474,40 @@ private int y(int a) {
                 193_428_131_138_340_668L) >>> 20) - 1;
     }
 
-    private void exponent(int e) {
-        append('E');
-        if (e < 0) {
-            append('-');
-            e = -e;
-        }
-        if (e < 10) {
-            appendDigit(e);
+    private void exponent(int ea) {
+        if (ea < 10) {
+            appendDigit(ea);
             return;
         }
         /*
          * For n = 2, m = 1 the table in section 10 of [1] shows
-         *     floor(e / 10) = floor(103 e / 2^10)
+         *     floor(ea / 10) = floor(103 ea / 2^10)
          */
-        int d = e * 103 >>> 10;
+        int d = ea * 103 >>> 10;
         appendDigit(d);
-        appendDigit(e - 10 * d);
+        appendDigit(ea - 10 * d);
+    }
+
+    private void appendSign() {
+        if (index != 0) {
+            bytes[0] = '-';
+        }
     }
 
     private void append(int c) {
-        bytes[++index] = (byte) c;
+        bytes[index++] = (byte) c;
     }
 
     private void appendDigit(int d) {
-        bytes[++index] = (byte) ('0' + d);
+        bytes[index++] = (byte) ('0' + d);
     }
 
-    /* Using the deprecated constructor enhances performance */
-    @SuppressWarnings("deprecation")
     private String charsToString() {
-        return new String(bytes, 0, 0, index + 1);
+        try {
+            return JLA.newStringNoRepl(bytes, StandardCharsets.ISO_8859_1);
+        } catch (CharacterCodingException e) {
```

Comment:
I want to know why newStringNoRepl throws CharacterCodingException instead of IllegalArgumentException. The place where I use it is forced to write redundant try_catch code.

Reply:
`newStringNoRepl` is originally intended to decode byte and throw an exception instead of putting a `?` when there is encoding error.  By chance, it does not copy the input array if the input array is compatible as backing storage.  As a result, we use it as "avoid copying" which has deviated from its initial purpose.